brave/brave-core
# [iOS] Add local search to Playlist

Resolves https://github.com/brave/brave-browser/issues/42437

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
- Add videos to playlist to multiple folders
- Open Playlist and tap the search button in the top right
- Verify you can search for items in multiple folders by name
- Verify items shown show the correct download states
- Verify long pressing on items shows the same context menu and that those buttons work correctly
- Verify that tapping on an item starts playing that item the same way as if tapped on the main sidebar/drawer list

## Screenshots

| a | b | c |
| --- | --- | --- |
| ![Simulator Screenshot - iPhone 16 Pro - 2024-12-09 at 12 29 00](https://github.com/user-attachments/assets/dfd09a81-d68e-421f-91e8-9a016d8fec6a) | ![Simulator Screenshot - iPhone 16 Pro - 2024-12-09 at 12 29 03](https://github.com/user-attachments/assets/c539e6bc-967c-4637-808d-95f86f1d0cfa) | ![Simulator Screenshot - iPhone 16 Pro - 2024-12-09 at 12 35 55](https://github.com/user-attachments/assets/f6e12b3b-2c48-41ad-855b-6a23a01aeb24) |
